File: LeetCode/LeetCode 75/1768. Merge Strings Alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int lenOne = word1.length();
        int lenTwo = word2.length();
        StringBuilder result = new StringBuilder();
        int i = 0, j = 0;

        while(i<lenOne || j < lenTwo) {
            if(i<lenOne) {
                result.append(word1.charAt(i++));
            }
            if(j < lenTwo) {
                result.append(word2.charAt(j++));
            }
        }
        return result.toString();
    }
}

File: Leetcode/Leetcode 75/1768. Merge Strings Alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let onePointer = 0, twoPointer = 0, oneLonger = word1.length > word2.length;
    let ans = '';
    while (onePointer < word1.length && twoPointer < word2.length) {
        ans += word1[onePointer];
        ans += word2[twoPointer];
        onePointer++;
        twoPointer++;
    }
    if(word1.length !== word2.length) {
        if (oneLonger) {
            ans += word1.substr(onePointer, word1.length);
        }
        else {
            ans += word2.substr(twoPointer, word2.length);
        }
    }
    return ans;
};
